src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip

```diff
@@ -0,0 +1,229 @@
+        {@preparations(api, method)}
+        {@method.callerResponseTypeName} response = {@sampleMethodCall(method)};
+
+        // Iterate over all response items, lazily performing RPCs as required
+        foreach ({@method.listMethod.resourceTypeName} item in response)
+        {
+            // Do something with each item
+            Console.Writeline(item);
+        }
+        
+        // Or iterate over fixed-sized pages, lazily performing RPCs as required
+        int pageSize = 10;
+        foreach (FixedSizePage<{@method.listMethod.resourceTypeName}> page in response.AsPages().WithFixedSize(pageSize))
+        {
+            // Do something with each page of items
+            Console.Writeline("A page of results:");
+            foreach ({@method.listMethod.resourceTypeName} item in page) {
+                Console.Writeline(item);
+            }
+        }
+        // End snippet
+    }
+@end
+
+@private pagedFlattenedMethodAsync(api, method)
+    public async Task {@method.name}()
+    {
+        // Snippet: {@method.name}({@snippetArgTypes(method.methodParams)},CallSettings)
+        // Additional: {@method.name}({@snippetArgTypes(method.methodParams)},CancellationToken)
+        {@preparations(api, method)}
+        {@method.callerResponseTypeName} response = await {@sampleMethodCall(method)};
+
+        // Iterate over all response items, lazily performing RPCs as required
+        foreach ({@method.listMethod.resourceTypeName} item in response)
+        {
+            // Do something with each item
+            Console.Writeline(item);
+        }
+        
+        // Or iterate over fixed-sized pages, lazily performing RPCs as required
+        int pageSize = 10;
+        foreach (FixedSizePage<{@method.listMethod.resourceTypeName}> page in response.AsPages().WithFixedSize(pageSize))
+        {
+            // Do something with each page of items
+            Console.Writeline("A page of results:");
+            foreach ({@method.listMethod.resourceTypeName} item in page) {
+                Console.Writeline(item);
+            }
+        }
+        // End snippet
+    }
+@end
+
+@private preparations(api, method)
+    // Create client
+    {@api.clientTypeName} {@method.apiVariableName} = {@api.clientTypeName}.Create();
+    // Initialize request argument(s)
+    {@topLevelInitCode(method.initCode)}
+    // Make the request
+@end
+
+@private snippetArgTypes(params)
+    @join param : params on ","
+        {@param.typeName}
+    @end
+@end
+
+@private sampleMethodCall(apiMethod)
+    {@apiMethod.apiVariableName}.{@apiMethod.name}({@sampleMethodCallArgList(apiMethod.initCode.fieldSettings)})
+@end
+
+@private sampleMethodCallArgList(fieldSettings)
+    @join fieldSetting : fieldSettings on ", "
+        {@fieldSetting.identifier}
+    @end
+@end
+
+@private topLevelInitCode(initCodeSpec)
+    @join line : initCodeSpec.topLevelLines
+        {@initCode(initCodeType(line).add(" ").add(line.identifier).add(" = "), line)};
```

Comment:
Could you figure out a different way of doing this? I'd rather have duplication in the initCode snippet, because it's hard to tell what's going on here.